### PR TITLE
[Typing][C-86] Add annotations for `python/paddle/incubate/operators/graph_khop_sampler.py`

### PR DESCRIPTION
**Pull request by inaomIIsfarell**
### PR Category

User Experience

### PR Types

Improvements

### Description

类型标注：

- python/paddle/incubate/nn/functional/variable_length_memory_efficient_attention.py

### Related links
 
- https://github.com/PaddlePaddle/Paddle/issues/65008

@SigureMo @megemini

### CODE REVIEW

File: python/paddle/incubate/nn/functional/variable_length_memory_efficient_attention.py
Comment (megemini) [inline python/paddle/incubate/nn/functional/variable_length_memory_efficient_attention.py:1]: C-80 不是提了 PR 了？https://github.com/PaddlePaddle/Paddle/pull/67095#issuecomment-2270667570

File: python/paddle/incubate/operators/graph_khop_sampler.py [hunk 1]
```diff
@@ -11,22 +11,28 @@
...
-    input_nodes,
-    sample_sizes,
-    sorted_eids=None,
-    return_eids=False,
-    name=None,
-):
+    row: Tensor,
+    colptr: Tensor,
+    input_nodes: Tensor,
+    sample_sizes: list[int] | tuple[int],
```
Comment (megemini) [inline python/paddle/incubate/operators/graph_khop_sampler.py:31]:
```suggestion
    sample_sizes: list[int] | tuple[int, ...],
```

File: python/paddle/incubate/operators/graph_khop_sampler.py [hunk 2]
```diff
@@ -11,22 +11,28 @@
...
+    sorted_eids: Tensor | None = None,
+    return_eids: bool = False,
+    name: str | None = None,
+) -> tuple[Tensor]:
```
Comment (megemini) [inline python/paddle/incubate/operators/graph_khop_sampler.py:35]: 使用 `overload`

File: python/paddle/incubate/nn/functional/variable_length_memory_efficient_attention.py
Comment (inaomIIsfarell) [reply python/paddle/incubate/nn/functional/variable_length_memory_efficient_attention.py:1]: 80和86我都重新开一个pr吧

**Pull request closed without merging**